### ISSUE TITLE
CA1508: False positive using Debug.Assert

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#4742]
```vb
    Public Sub Main(args As String())
        Debug.Assert(args.Length = 0)
        If args.Length = 0 Then
        End If
    End Sub
```

![image](https://user-images.githubusercontent.com/31348972/106001971-3f2c0d80-60b9-11eb-92eb-34edf3b7255c.png)

The condition can still be reachable in release builds.

This is related to most (if not all) diagnostics by this rule produced on project-system.

FYI @drewnoakes

**Comment by mavasani** [comment]
This is by-design. The analyzer is helping you avoid functional issues by identifying that you have both a debug assert and a runtime check for the exact same thing. You should probably refactor the code as below to improve debug build performance:

```vb
Public Sub Main(args As String())
        If args.Length = 0 Then
        Else
            Debug.Fail("message")
        End If
End Sub
```
